src/url.js

```diff
@@ -95,6 +95,18 @@ export function assertHttpsUrl(urlString, elementContext) {
   return urlString;
 }
 
+/**
+ * Asserts that a given url is an absolute HTTP or HTTPS URL.
+ * @param {string} urlString
+ * @return {string}
+ */
+export function assertAbsoluteUrl(urlString) {
+  assert(/^(http\:|https\:)/i.test(urlString),
```

Comment:
Do a parse and then compare the scheme to http or https. You parse after anyway. That is more secure, because it is sure the browser will actually interpret it as such.